### DISCUSSION TITLE
How to allow only versions below a specific minor release when a package has a suffix

### DISCUSSION THREAD

**Discussion by regressEdo** [renovatebot/renovate#28170] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

We are currently running `php` for some of our applications and inside the dockerfile we currently use this version:

`php:8.1.27-fpm`

the problem we are currently facing is that Renovate with this global configuration it's not able to open any new mr:

`"packageRules": [`
`    {`
`        "matchDatasources": ["docker"],`
`        "matchPackageNames": ["php"],`
`        "allowedVersions": "<= 8.2.0",`
`        "separateMinorPatch": true`
`    }`
`],`

the problem is related to the suffix `-fpm`

we tried also this configuration:

`"packageRules": [`
`    {`
`        "matchDatasources": ["docker"],`
`        "matchPackageNames": ["php"],`
`        "separateMinorPatch": true`
`    }`
`],`

In this case the new mr get opened with the proper suffix maintained but of course the `allowedVersion` property doesn't work anymore, so we also get updates for `php 8.3` which we don't want

ultimatly we tried this configuration:

`"packageRules": [`
`    {`
`        "matchDatasources": ["docker"],`
`        "matchPackageNames": ["php"],`
`        "allowedVersions": "<= 8.2.0",`
`        "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(-(?<compatibility>.*))?$",`
`        "separateMultipleMinor": true`
`    }`
`],`

this configuration respects the ` allowedVersions` paramenter but it doesn't keep track of the `patch` release version, from the original patckage it tries to update it this way: `php:8.1.27-fpm -> php:8.2-fpm`

I'm probably missing a simple configuration here but I cannot find it, please point me in the right direction!

Thanks

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Try experimenting with versionCompatibility and separating the suffix as the compatibility part

**Reply by regressEdo** [reply]
Perfect, it worked out prefectly by just adding:

`"versionCompatibility": "^(?<version>[^-]+)(?<compatibility>-.*)?$",`
